src/main/java/org/apache/ibatis/builder/annotation/MapperAnnotationBuilder.java

```diff
@@ -124,20 +83,30 @@ public void parse() {
         if (!canHaveStatement(method)) {
           continue;
         }
-        if (getAnnotationWrapper(method, false, Select.class, SelectProvider.class).isPresent()
-            && method.getAnnotation(ResultMap.class) == null) {
-          parseResultMap(method);
-        }
         try {
+          if (getAnnotationWrapper(method, false, Select.class, SelectProvider.class).isPresent() &&
+            method.getAnnotation(ResultMap.class) == null)
+            parseResultMap(method);
           parseStatement(method);
         } catch (IncompleteElementException e) {
           configuration.addIncompleteMethod(new MethodResolver(this, method));
+          throw new IncompleteElementException(getParsingErrorMessage(method), e);
+        } catch (Exception e) {
+          throw new BuilderException(getParsingErrorMessage(method), e);
```

Comment:
I realize any callers catching an IllegalStateException or other exception would need to be updated. An alternative approach to add the method context to the exception would be to reflectively create the a new instance of the exception caught here, add the context, but may be overkill here. The reflective exception class may throw an exception in it's constructor and the resulting code would be less readable because it would need to handle a 1% case.